src/lighteval/main_baseline.py

```diff
@@ -20,33 +20,66 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 
-from lighteval.logging.evaluation_tracker import EvaluationTracker
-from lighteval.metrics.utils.metric_utils import MetricCategory
-from lighteval.models.abstract_model import ModelInfo
-from lighteval.tasks.lighteval_task import LightevalTask
-from lighteval.tasks.registry import Registry, taskinfo_selector
-from lighteval.utils.utils import as_list
 
-
-def main(args):
+import os
+from typing import Optional
+
+from typer import Argument, Option
+from typing_extensions import Annotated
+
+
+CACHE_DIR: str = os.getenv("HF_HOME", "/scratch")
+
+HELP_PANNEL_NAME_1 = "Common Paramaters"
+HELP_PANNEL_NAME_2 = "Logging Parameters"
+HELP_PANNEL_NAME_3 = "Debug Paramaters"
+HELP_PANNEL_NAME_4 = "Modeling Paramaters"
+
+
+def baseline(
+    tasks: Annotated[str, Argument(help="Comma-separated list of tasks to evaluate on.")],
+    cache_dir: Annotated[
+        str, Option(help="Cache directory for datasets and models.", rich_help_panel=HELP_PANNEL_NAME_1)
+    ] = CACHE_DIR,
+    custom_tasks: Annotated[
+        Optional[str], Option(help="Path to custom tasks directory.", rich_help_panel=HELP_PANNEL_NAME_1)
+    ] = None,
+    dataset_loading_processes: Annotated[
+        int, Option(help="Number of processes to use for dataset loading.", rich_help_panel=HELP_PANNEL_NAME_1)
+    ] = 1,
+    output_dir: Annotated[
+        str, Option(help="Output directory for evaluation results.", rich_help_panel=HELP_PANNEL_NAME_2)
+    ] = "results",
+    max_samples: Annotated[
+        Optional[int], Option(help="Maximum number of samples to evaluate on.", rich_help_panel=HELP_PANNEL_NAME_3)
+    ] = None,
+):
     """
     Compute baselines for given tasks.
 
     It has been tested with generative and accuracy tasks, but may not work correctly for other task types.
 
     The baseline is computed as follows:
+
     - For multiple-choice tasks: It assumes random guessing, so the score is n_correct/number_of_choices.
     - For other metrics: It assigns a score of 0, which may not be appropriate for all task types.
 
     Note:
         This baseline computation may not be suitable for all task types and should be used with caution.
     """
-    task_registry = Registry(cache_dir=args.cache_dir, custom_tasks=args.custom_tasks)
-    task_names_list, fewshots_dict = taskinfo_selector(args.tasks, task_registry)
+    from lighteval.logging.evaluation_tracker import EvaluationTracker
+    from lighteval.metrics.utils.metric_utils import MetricCategory
+    from lighteval.models.abstract_model import ModelInfo
+    from lighteval.tasks.lighteval_task import LightevalTask
+    from lighteval.tasks.registry import Registry, taskinfo_selector
+    from lighteval.utils.utils import as_list
```

Comment:
ooc, why do you do imports inside the class for lighteval related code?

Reply:
Lighteval imports can be quite expensive. This allow to import only what;s relevant for our need when calling a command.
For example `lighteval --help` is very fast thanks to this